core/state/node/schema.go

```diff
@@ -63,12 +63,20 @@ func (s NodeState) Bytes() ([]byte, error) {
 }
 
 func (s NodeState) GetAppByID(appID string) (*AppInstallationState, error) {
+	app, ok := s.AppInstallations[appID]
+	if !ok {
+		return nil, fmt.Errorf("app with ID %s not found", appID)
+	}
+	return app, nil
+}
+
+func (s NodeState) GetAppByName(appName string) (*AppInstallationState, error) {
 	for _, app := range s.AppInstallations {
-		if app.ID == appID {
+		if app.Name == appName {
```

Comment:
a later question, but what if there are multiple? we probably don't need to use this function except for showing UI stuff right?

Reply:
This is leftover, but not currently used, I can remove it.